core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -222,26 +222,40 @@ public InternalEvent process(InternalEvent event) throws MuleException {
     initialiseIfInitialisable(pipeline);
   }
 
+  /*
+   * Processor that dispatches incoming source Events to the internal pipeline the Sink. The way in which the Event is dispatched
+   * and how overload is handled depends on the Source back-pressure strategy.
+   */
   private ReactiveProcessor dispatchToFlow(Sink sink) {
     if (source.getBackPressureStrategy() == WAIT) {
-      return publisher -> from(publisher).doOnNext(event -> sink.accept(event))
+      // If back-pressure strategy is WAIT then using blocking `accept(Event event)` to dispatch Event
+      return publisher -> from(publisher)
+          .doOnNext(event -> {
+            try {
+              sink.accept(event);
+            } catch (RejectedExecutionException ree) {
+              MessagingException me = new MessagingException(event, ree, this);
+              event.getContext().error(exceptionResolver.resolve(me, getMuleContext()));
+            }
+          })
           .flatMap(event -> Mono.from(event.getContext().getResponsePublisher()));
     } else {
+      // If back-pressure strategy is FAIL/DROP then using back-pressure aware `accept(Event event)` to dispatch Event
       return publisher -> from(publisher).flatMap(event -> {
         if (sink.emit(event)) {
           return event.getContext().getResponsePublisher();
         } else {
           if (source.getBackPressureStrategy() == DROP) {
+            // If Event is not accepted and the back-pressure strategy is DROP then drop Event.
             return empty();
```

Comment:
should we at leas log a warning when an event is dropped? Having the id of the dropped event may help troubleshooting

Reply:
If a source chooses to use DROP, then it's because it's processing a continuous stream of data such as temperatures and it is valid for the use case to sample data rather than process all data points. As such I don't think it makes sense to add any logging here.